data_lang/utf8.h

```diff
@@ -0,0 +1,178 @@
+#ifndef DATA_LANG_UTF8_H
+#define DATA_LANG_UTF8_H
+
+#include <stdio.h>
+
+#include <stdint.h> // uint32_t
+#include <stddef.h> // size_t
+
+#ifdef __cplusplus
```

Comment:
hm what breaks if we don't have this?  I don't recall using `__cplusplus` in our codebase

`j8.h` is shared between C++ and CPython C, and it seems to work

Reply:
Yeah, we don't need that here. All the code in this file fits within the shared subset of C and C++.